libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -493,8 +493,16 @@ bool AP_Arming::arm_checks(uint8_t method)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_LOGGING)) {
-        if (!DataFlash_Class::instance()->logging_started()) {
+        // allow logging to start by indicating (falsely) that the vehicle is armed:
+        DataFlash_Class *df = DataFlash_Class::instance();
+        df->set_vehicle_armed(true);
```

Comment:
A couple of ideas that did come to me:
 - post-arming checks which would activate before we set any notify flags
   - you now have race conditions against other threads which may or may not care about your arming state
 - lists of cleanup handlers in the case an arming check fails.
 - another entry point to DataFlash to say "do everything you would do if vehicle was just set to armed (I *really* don't like this option...)
 - restructure to have a `goto BAD_LOGGING:` set of cleanup handlers

I didn't act on any of those, obviously.